### DISCUSSION TITLE
Open-source standalone library for runtime warnings?

### DISCUSSION THREAD

**Discussion by sebj** [pointfreeco/swift-composable-architecture#1436] [General]
The runtime warnings that Composable Architecture is able to generate are super neat. I can see the code for this is contained within [`RuntimeWarnings.swift`](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Sources/ComposableArchitecture/Internal/RuntimeWarnings.swift).

Is there any likelihood that this would be open-sourced as a standalone library?

**Comment by mbrandonw** [comment]
It's something we have discussed a number of times, and we are thinking it will happen eventually. We even had to copy-paste it to another [library](https://github.com/pointfreeco/xctest-dynamic-overlay/blob/main/Sources/XCTestDynamicOverlay/Internal/RuntimeWarnings.swift). But, we're not sure _when_ we will do it, so for now if you want similar functionality you should probably copy-paste that file into your project.
